src/dmd/dmodule.d

```diff
@@ -506,7 +508,10 @@ extern (C++) final class Module : Package
         /* Look for the source file
          */
         if (const result = lookForSourceFile(filename))
+        {
+            scope(exit) FileName.free(result.ptr);
```

Comment:
remove `scope(exit)` and move the rest to after line 513. DMD doesn't use exceptions, so this wouldn't be a problem.